CRM/Mailing/BAO/Mailing.php

```diff
@@ -1409,7 +1409,11 @@ public function &getGroupNames() {
    * @return CRM_Mailing_DAO_Mailing
    */
   public static function add(&$params, $ids = array()) {
-    $id = CRM_Utils_Array::value('mailing_id', $ids, CRM_Utils_Array::value('id', $params));
+    $id = CRM_Utils_Array::value('id', $params, CRM_Utils_Array::value('mailing_id', $ids));
+
+    if (empty($params['id']) && !empty($ids)) {
```

Comment:
Shouldn't this condition just be `if (!empty($ids))`?

Reply:
$ids actually won't be empty coming from the api - it will hold ['Mailing' => $params['id']]  - but the goal is to make the $ids array first meaningless & then to remove it - as long as they are correctly setting the $params array they can pass anything or nothing in $ids